src/components/HappHosting/HappHosting.connector.js

```diff
@@ -24,20 +23,30 @@ const hostingUser = graphql(HostingUserQuery, {
   props: ({ data: { hostingUser } }) => ({ hostingUser })
 })
 
-const registerHostingUser = graphql( RegisterHostingUserMutation, {
+const registerHostingUser = graphql(RegisterHostingUserMutation, {
   props: ({ mutate }) => {
     return {
-      // NOTE: Currently host_doc is not validated and the content is irrelevant to the dna...
-      registerHostingUser: ({host_doc}) => mutate({
+      // NOTE: Currently hostDoc is not validated and the content is irrelevant to the dna...
+      registerHostingUser: hostDoc => mutate({
         variables: {
-          host_doc
+          hostDoc
+        },
+        update: (cache, { data: { registerHostingUser } }) => {
```

Comment:
this whole update function is the thing I was talking about "the last thing you haven't seen about Apollo". Basically, it's a callback that gets called after the mutation returns. it gets passed the result of the mutation, as well as the cache. The cache is the Apollo equivalent of the redux store, it's where all the local state is stored, specifically the results of all our graphQL queries.

So what I'm doing here is taking the result of the registerHostingUser mutation and writing it to the cache under the HostingUserQuery, so that when Apollo then goes to fetch the results of HostingUserQuery, it will return the result of registerHostingUser, and won't have to call the dna again to get the result.

Most often we get this kind of behavior for free, but if we don't already have an id for something (ie, when we're creating it), we have to do it manually like this.

This is a very dense thing to try and communicate in this little comment box, so if I didn't make it clear enough, just let me know!